cmake/SundialsBuildOptionsPre.cmake

```diff
@@ -76,12 +76,12 @@ sundials_option(SUNDIALS_BUILD_WITH_PROFILING BOOL "${DOCSTR}" OFF)
 # ---------------------------------------------------------------
 
 set(DOCSTR "Build with logging capabilities enabled (0 = no logging, 1 = errors, 2 = +warnings, 3 = +info, 4 = +debug, 5 = +extras")
-sundials_option(SUNDIALS_LOGGING_LEVEL STRING "${DOCSTR}" 0
+sundials_option(SUNDIALS_LOGGING_LEVEL STRING "${DOCSTR}" 3
```

Comment:
I think info level is the right level but we need to be careful what we put in the info level (this is why I changed a lot of the ARKODE output to `DEBUG` level). If we switch only do errors and warnings, then things that previously were always accessible, like the nonlinear solver iteration info would not be usable by default.

Reply:
Also, the info level has no additional output compared to what we currently have in v6 now that I changed most (or all) of the ARKODE diagnostic output to be at the debug level.

Reply:
Would there still be additional function call overhead with level `3` vs level `2` even if it's not written out? If we use the info level for getting the history of statistics over time then this could also involve additional conditionals for what needs to be output.

Reply:
The diagnostic output in ARKODE is guarded with an ifdef that checks the level, so no there is no extra overhead for those. If you left out the ifdef's then, yes, there would still be the additional conditionals. There should be zero overhead between level 2 and 3 with _this_ branch.  What we add to level 3 in the future could introduce overhead, of course, which is why I said we will want to carefully consider what is "INFO" and what is for debugging. 

I am not sure what exactly you mean by a "history of statistics over time", but if/when we go to implement it, we would want to consider if it should be at the INFO level or DEBUG level.

There are two downsides to choosing level `2` as the default:
1. KINSOL's long-standing info functions which would not be usable by default
2. The newer (introduced maybe in version 5?) SUNNonlinearSolver and SUNLinearSolver "info" functions would not be usable by default. 
Users of these functions would have to choose level `3`, so defaulting to `2` may break things for them. 

Since we are doing a major release, I suppose thats fine. We just need to make sure that we are OK with it. If we are, then level `2` should be fine as the default.

Reply:
By "history of statistics over time" I mean things like step size history, nonlinear/linear iterations per step, nonlinear/linear residuals per iteration, etc. To me these seem like output that would live at the INFO level as they're useful for understanding the behavior/performance of a method. This is potentially more output than we would want enabled be default and is why I'm thinking a default of `2`. This would break the two cases you note and that's something we should discuss. 

For now I think this PR can leave the default unchanged so we can discuss further without delaying the other changes in the PR.

Reply:
By "this PR can leave the default unchanged" do you mean leaving it as `0` or leaving it as-is in the PR right now (`3`) ?

Reply:
I mean leave it as it was before, `0`

Reply:
I think if we are going to do that, might as well make it at least `2`. It has to be at least `2` for the error handling.  Ill switch it to `2` for now, but we will need to discuss it on Tuesday.